python/cudf/cudf/core/join/casting_logic.py

```diff
@@ -0,0 +1,282 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.
+
+import warnings
+
+import numpy as np
+import pandas as pd
+
+import cudf
+from cudf.core.dtypes import CategoricalDtype
+
+
+def _input_to_libcudf_castrules_both_cat(lcol, rcol, how):
+    """
```

Comment:
Should these rules be documented somewhere public?

Reply:
yes! It definitely should be. Let me see about putting that together.

Reply:
Added this to the `DataFrame.merge` docstring